app/controllers/foreman_tasks/api/recurring_logics_controller.rb

```diff
@@ -39,9 +39,22 @@ def resource_class
         ForemanTasks::RecurringLogic
       end
 
+      api :POST, '/recurring_logics/bulk_destroy', N_('Delete recurring logics by search query')
+      param :search, String, :desc => N_('Search query'), :required => true
+      def bulk_destroy
+        if params[:search].nil?
```

Comment:
How about using `blank?` instead of `nil?` so it also catches empty strings?

Reply:
I thought we wanted to keep the option to destroy all by sending an empty string?

Reply:
People will still be able to destroy all by explicitly matching on all with something like `state ~ *`

Reply:
Changed to blank and changed the argument error to be `BadRequest`